internal/tools/utility/alloydbwaitforoperation/alloydbwaitforoperation.go

```diff
@@ -251,9 +242,15 @@ func (t *Tool) Invoke(ctx context.Context, params tools.ParamValues) (any, error
 
 		req, _ := http.NewRequest(http.MethodGet, urlString, nil)
 
-		for k, v := range t.Headers {
-			req.Header.Set(k, v)
+		tokenSource, err := google.DefaultTokenSource(ctx, "https://www.googleapis.com/auth/cloud-platform")
```

Comment:
We should note the specific permissions needed and how this is authenticated through environment credentials